File: ios/ShortDrama/Sources/Domain/Entities/BookingAssetSummary.swift
import Foundation

struct BookingAssetSummary: Equatable, Sendable {
    let onlineCount: Int
    let upcomingCount: Int

    static let empty = BookingAssetSummary(onlineCount: 0, upcomingCount: 0)

    func count(for status: BookingAssetAvailabilityStatus) -> Int {
        switch status {
        case .online:
            return onlineCount
        case .upcoming:
            return upcomingCount
        }
    }
}

File: android/app/src/main/java/com/djs66256/short_drama/domain/model/BookingAssetSummary.kt
package com.djs66256.short_drama.domain.model

data class BookingAssetSummary(
    val onlineCount: Int = 0,
    val upcomingCount: Int = 0,
) {
    fun countFor(status: BookingAssetStatus): Int = when (status) {
        BookingAssetStatus.ONLINE -> onlineCount
        BookingAssetStatus.UPCOMING -> upcomingCount
    }
}
